recipes/cuda-cccl/meta.yaml

```diff
@@ -0,0 +1,62 @@
+{% set name = "cuda-cccl" %}
+{% set version = "12.0.90" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/cuda_cccl/{{ platform }}/cuda_cccl-{{ platform }}-{{ version }}-archive.{{ extension }}
+  sha256: 37f6e03b6f95579d8e2e96b9ba15eeb1224dcd18b21ad50c3ac249e408152a24  # [linux64]
+  sha256: c6de67acd6a65b24db0383f6330b842ed19b6c27a27a37f828e9f3e96cc96f72  # [ppc64le]
+  sha256: f2a19927c558a9c872fa87eda4c0dc0446e5c4c744df4f846d01f4d4e08c8cd7  # [aarch64]
+  sha256: dd422d4cc8462d112411659a56cc78cb6010d704f83db3e48c55a3f66eed5ec2  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+
+outputs:
+  - name: cuda-cccl
+    requirements:
+      run:
+        - cuda-cccl_{{ target_platform }} =={{ version }}
```

Comment:
Where is this package being defined? Why do we need platform specific packages for cccl?

Reply:
It is the second output of this recipe. It is needed for the `cuda-cudart-dev` and `cuda-nvcc` target specific packages.

Reply:
Why couldn't they just depend on `cuda-cccl` instead of `cuda-cccl_{{ target_platform }}`? I don't fully understand why target specific package names are needed but they all seem to be connected to where there's special jinja templates used. Maybe @jakirkham knows more?

Reply:
This crossed my mind as well, but I think it's probably good practice to align with how the `https://developer.download.nvidia.com/` distributes CCCL: it's provided as a platform-specific archive.

Reply:
Yes, I'm not questioning it being platform-specific, and we could use platform-specific sources, but you don't need separate package names for that. Most packages that have a binary component are platform-specific, but they do so under a certain name. The ones that use the `_{{ target_platform }}` approach are ones that sit behind special templates like `{{ compiler('cxx') }}`. I don't fully understand the reasoning behind this but maybe someone with more understanding could explain.

Reply:
Good point. We could use selectors to achieve the platform-specific source. I think I agree that the `_{{ target_platform }}` package is unnecessary. @adibbley @leofang What are your thoughts?

Reply:
> The ones that use the _{{ target_platform }} approach are ones that sit behind special templates like {{ compiler('cxx') }}

That's exactly the reason to use a platform specific name. To allow the cuda compiler to cross compile.

Reply:
Okay, so if I understand correctly we then require this package to be platformed as well for cross compilation purposes?

Reply:
Yes

Reply:
That sounds right. Thanks for the correction @isuruf.

Reply:
Great, resolving this.